webui/react/src/pages/ExperimentDetails/ExperimentCheckpoints.tsx

```diff
@@ -341,31 +331,35 @@ const ExperimentCheckpoints: React.FC<Props> = ({ experiment, pageRef }: Props)
           onAction={(action) => submitBatchAction(action)}
           onClear={clearSelected}
         />
-        <InteractiveTable
-          columns={columns}
-          containerRef={pageRef}
-          ContextMenu={CheckpointActionDropdown}
-          dataSource={checkpoints}
-          loading={isLoading}
-          pagination={getFullPaginationConfig(
-            {
-              limit: settings.tableLimit,
-              offset: settings.tableOffset,
-            },
-            total,
-          )}
-          rowClassName={defaultRowClassName({ clickable: false })}
-          rowKey="uuid"
-          rowSelection={{
-            onChange: handleTableRowSelect,
-            preserveSelectedRowKeys: true,
-            selectedRowKeys: settings.row ?? [],
-          }}
-          settings={settings}
-          showSorterTooltip={false}
-          size="small"
-          updateSettings={updateSettings as UpdateSettings<InteractiveTableSettings>}
-        />
+        {settings ? (
+          <InteractiveTable
+            columns={columns}
+            containerRef={pageRef}
+            ContextMenu={CheckpointActionDropdown}
+            dataSource={checkpoints}
+            loading={isLoading}
+            pagination={getFullPaginationConfig(
+              {
+                limit: settings.tableLimit ?? 0,
+                offset: settings.tableOffset ?? 0,
```

Comment:
`?? 0` means we cant fetch anything if `tableLimit` is undefined. I think we should find other way to treat this